actors/market/src/state.rs

```diff
@@ -255,6 +255,26 @@ impl State {
         find_proposal(&self.load_proposals(store)?, deal_id)
     }
 
+    pub fn find_proposal_from_cache<BS: Blockstore>(
```

Comment:
These probably shouldn't be on `&self` because that's not useful here. (I'm not the expert here, but) I think I would just do this inline since `get()` is all you need to be doing and transforming the error code is fairly straightforward. You could also consider attempting to reuse the static `get_proposal`, but its checking of `next_id` is probably beyond what we need here.

Same comment applies for the function below for the deal, making this a method on state that takes an element of the state is not quite the right pattern here. They could either be static helper functions or just do it inline when it's realtively simple like this.

Reply:
Thank you for the feedback. I understand that making these methods instance methods on &self might not be the best pattern since they do not utilize the instance state. I've used inline get() where they are used, as suggested.